runtime/lua/vim/secure.lua

```diff
@@ -120,48 +122,66 @@ end
 --- $XDG_STATE_HOME/nvim/trust.
 ---
 --- @param path (string) Path to a file to update status for.
---- @param allow (boolean) If true, set status to allow, otherwise set to deny.
-function M.trust(path, allow)
+--- @param mode (string) One of the following:
+---   - 'allow': Add file to trust database and set it as trusted
+---   - 'deny': Add file to trust database and set it as denied
+---   - 'forget': Remove file from trust database
+--- @return (boolean, string|nil) success, errmsg: true and nil if the operation was successful,
+---   otherwise false and an error message.
+function M.trust(path, mode)
   vim.validate({ path = { path, 's' } })
-  vim.validate({ allow = { allow, 'b' } })
+  vim.validate({ allow = { mode, 's' } })
+  if mode ~= 'allow' and mode ~= 'deny' and mode ~= 'forget' then
+    return false, string.format('invalid mode: %s', mode)
```

Comment:
In this case since we’re validating the input arguments, let’s use a real `error` rather than returning a string. Alternatively use an `assert` or put it in the `vim.validate` call.

Reply:
Resolved in dc8040d5ecd62fb3235ac730d2e744727c0c6bc1